api/src/controller/user-controller.ts

```diff
@@ -183,19 +183,46 @@ export class UserController {
         response: Response,
         next: NextFunction
     ): Promise<Response> {
-        const userID: number = Number(request.params.userID);
-        const newPassword: string = request.body.password;
+        const userID: string = request.params.userID;
+        const userIDasNumber: number = Number(userID);
+        const verifiedUser = request.body.verifiedUser;
+        const newPassword: string = request.body.newPassword;
+        const oldPassword: string = request.body.oldPassword;
+        let userPasswordHash: string;
 
         if (newPassword === undefined) {
             return response.status(404).send({ status: "Arguments missing" });
         }
 
+        // If requesting user is not an admin, he must provide the old password
+        if (!verifiedUser.isAdmin) {
+            try {
+                userPasswordHash =
+                    await UserService.getUserPasswordHashByUserID({
+                        userID,
+                    });
+            } catch (error) {
+                return response.status(404).send({ status: "User not found" });
```

Comment:
It's not possible to scan users this way, because as a non admin user you can only connect to this endpoint for your own user id. This error should never occur as a normal user. It should only happen as an admin user and admins can legally see als users.